packages/auth/src/signature/middleware.test.ts

```diff
@@ -401,6 +405,55 @@ describe('Signature Service', (): void => {
       scope.isDone()
     })
 
+    test('middleware succeeds if BYPASS_SIGNATURE_VALIDATION is true with bad signature', async (): Promise<void> => {
+      const altDeps = await initIocContainer({
+        ...Config,
+        bypassSignatureValidation: true
+      })
+
+      const altContainer = await createTestApp(altDeps)
+      appContainers.push(altContainer)
```

Comment:
It might also work to keep `altContainer` **contain**ed to this test, call `shutdown` within the test (maybe in a `try`/`finally`), and remove `appContainers`.

Reply:
That would be nice, but that makes the following test fail with:

```
   Unable to acquire a connection
```

(from knex)

I dug into it a little and the shutdown calls `knex.destroy()` 

https://github.com/interledger/rafiki/blob/main/packages/auth/src/tests/app.ts#L47-L49

https://github.com/interledger/rafiki/blob/3807051d636f0c523d8f885c3d71b91d04cbc4c4/packages/auth/src/index.ts#L144

So the current approach works around this by keeping the test connections open until done and then closing them. The container system could get a refactor but I'm not sure it's worth it.

Reply:
Thanks for checking :vulcan_salute: